### PR TITLE
add mvp in app

### PR DESCRIPTION
**Pull request by pipi3gunD**

### CODE REVIEW

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/AlertDialogInfo.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import android.app.Activity;
+import android.app.Dialog;
+import android.view.View;
+import android.view.Window;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.eduard.caloriecounter.R;
+
+public class AlertDialogInfo {
```
[the remark this answers is not in the document]
Comment (Vladus177) [reply app/src/main/java/com/eduard/caloriecounter/presentation/fragments/AlertDialogInfo.java]: For example:
`public abstract class BaseFragment extends Fragment {
  
    private BasePresenter basePresenter;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        onInjection();
        onPrepareFragment(view);
        onPreparePresenter();
    }

    protected void onPreparePresenter() {

    }

    protected void onPrepareFragment(@NonNull View view) {

    }

    protected void onInjection() {

    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        releasePresenter();
    }

    public void showToast(String message) {
        Toast.makeText(getActivity(),
                message,
                Toast.LENGTH_SHORT).show();
    }

    public void showInformationDialog(String information) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setMessage(information)
                .setCancelable(false)
                .setNeutralButton("OK", (dialog, which) -> dialog.dismiss());
        AlertDialog alert = builder.create();
        alert.show();
    }

    public void attachPresenter(BasePresenter basePresenter, BasePresenter.View view) {
        this.basePresenter = basePresenter;
        this.basePresenter.attachView(view);
    }

    private void releasePresenter() {
        if (basePresenter != null) {
            basePresenter.release();
        }
    }
}`

File: app/src/main/res/layout/collection_of_info_fragment.xml [hunk 2]
```diff
@@ -1,78 +1,86 @@
...
-            </Button>
 
-        </LinearLayout>
+        <TextView
+            android:id="@+id/tv_infoWeight_test"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="10dp"
+            android:text="INFOO"
+            android:textSize="30sp"
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/res/layout/collection_of_info_fragment.xml]: ready

File: app/src/main/res/layout/collection_of_info_fragment.xml [hunk 3]
```diff
@@ -1,78 +1,86 @@
...
-                android:layout_marginRight="50dp">
-            </Button>
 
-        </LinearLayout>
+        <TextView
+            android:id="@+id/tv_infoWeight_test"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="10dp"
+            android:text="INFOO"
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/res/layout/collection_of_info_fragment.xml]: ready

File: app/src/main/res/layout/collection_of_info_fragment.xml [hunk 4]
```diff
@@ -1,78 +1,86 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+android:layout_width="match_parent"
+android:layout_height="match_parent"
+android:orientation="vertical">
+
+<LinearLayout
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/res/layout/collection_of_info_fragment.xml]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+
+    public interface View{
+
+        void updateUserInfoTextView(String info);
+        void showProgressBar();
+        void hideProgressBar();
+
+    }
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 6]
```diff
@@ -1,95 +1,161 @@
...
+
+                Toast toast = Toast.makeText(this, presenter.updateResult() + "",Toast.LENGTH_LONG);
+                toast.show();
             }
         });
+    }
+
+    private void initProgressBar() {
+        progressBar = new ProgressBar(this, null, android.R.attr.progressBarStyleSmall);
+        progressBar.setIndeterminate(true);
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 7]
```diff
@@ -1,95 +1,161 @@
...
+        btnGenerate = findViewById(R.id.btn_generate);
+        btnGenerate.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onClick(View v) {
-                dialog.dismiss();
+            public void onClick(View view) {
+
+                hideProgressBar();
+
+                Toast toast = Toast.makeText(this, presenter.updateResult() + "",Toast.LENGTH_LONG);
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 8]
```diff
@@ -1,95 +1,161 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.collection_of_info_fragment);
 
-//        showDialog(this,"ochen' vaznaya infa");
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
 
+        presenter = new CollectionPresenter(this);
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: this is a temporary option that will be replaced by dagger2

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 9]
```diff
@@ -1,95 +1,161 @@
...
+
+public class CollectionFragment extends AppCompatActivity implements CollectionPresenter.View {
+
+//    @Inject
+//    AlertDialogInfo alertDialogInfo;
+//    @Inject
+//    CollectionPresenter collectionPresenter;
+
+    private CollectionPresenter presenter;
+    private TextView textViewInfoW;
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/activity/MainActivity.java [hunk 10]
```diff
@@ -23,14 +21,13 @@ protected void onCreate(Bundle savedInstanceState) {
         btnStart.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                //начало конструкции
                 try {
                     Intent intent = new Intent(MainActivity.this, CollectionFragment.class);
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/activity/MainActivity.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/AlertDialogInfo.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/fragments/AlertDialogInfo.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java [hunk 11]
```diff
@@ -0,0 +1,98 @@
+package com.eduard.caloriecounter.presentation.presenter;
+
+import com.eduard.caloriecounter.presentation.model.User;
+
+public class CollectionPresenter {
```
[the remark this answers is not in the document]
Comment (pipi3gunD) [reply app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java]: ready

**Pull request merged**